src/Microsoft.OData.Core/JsonLight/ODataJsonLightContextUriParser.cs

```diff
@@ -71,19 +71,51 @@ private ODataJsonLightContextUriParser(IEdmModel model, Uri contextUriFromPayloa
             ODataPayloadKind payloadKind,
             Func<IEdmType, string, IEdmType> clientCustomTypeResolver,
             bool needParseFragment,
-            bool throwIfMetadataConflict = true)
+            bool throwIfMetadataConflict = true,
+            Uri baseUri = null,
+            IEdmNavigationSource navigationSource = null)
         {
             if (contextUriFromPayload == null)
             {
                 throw new ODataException(ODataErrorStrings.ODataJsonLightContextUriParser_NullMetadataDocumentUri);
             }
 
-            // Create an absolute URI from the payload string
-            // TODO: Support relative context uri and resolving other relative uris
+            // Create a context URI from the payload string.
             Uri contextUri;
             if (!Uri.TryCreate(contextUriFromPayload, UriKind.Absolute, out contextUri))
             {
-                throw new ODataException(ODataErrorStrings.ODataJsonLightContextUriParser_TopLevelContextUrlShouldBeAbsolute(contextUriFromPayload));
+                if (baseUri != null)
+                {
+                    ODataUri oDataUri = new ODataUri() { ServiceRoot = baseUri };
+
+                    // This caters for this format: #$delta.
+                    if (contextUriFromPayload.Equals(ODataConstants.HashDeltaResourceSet, StringComparison.OrdinalIgnoreCase))
+                    {
+                        if (string.IsNullOrEmpty(navigationSource?.Name))
+                        {
+                            throw new ODataException(ODataErrorStrings.ODataJsonLightContextUriParser_InvalidContextUrl(contextUriFromPayload));
+                        }
+                        else
+                        {
+                            contextUriFromPayload = oDataUri.MetadataDocumentUri.ToString() + ODataConstants.ContextUriFragmentIndicator + navigationSource.Name + ODataConstants.UriSegmentSeparator + ODataConstants.DeltaResourceSet;
+                        }       

```

Comment:
The `else` keyword isn't necessary because the method execution will stop if the exception in the if statement is thrown. Consider removing the unnecessary indentation:
```suggestion
                        if (string.IsNullOrEmpty(navigationSource?.Name))
                        {
                            throw new ODataException(ODataErrorStrings.ODataJsonLightContextUriParser_InvalidContextUrl(contextUriFromPayload));
                        }
                        
                        contextUriFromPayload = oDataUri.MetadataDocumentUri.ToString() + ODataConstants.ContextUriFragmentIndicator + navigationSource.Name + ODataConstants.UriSegmentSeparator + ODataConstants.DeltaResourceSet;

```